doc/dev/templates.md

```diff
@@ -0,0 +1,143 @@
+# Istio Mixer: Template Developer’s Guide
+
+Templates are a foundational building block of the Mixer architecture. This document
+describes the template format and how to extend Mixer with custom templates.
+
+- [Background](#background)
+- [Template format](#template-format)
+- [Supported field names](#supported-field-names)
+- [Supported field types](#supported-field-types)
+- [Adding a template to Mixer](#adding-a-template-to-mixer)
+- [Template evolution compatibility](#template-evolution-compatibility)
+
+## Background
+
+Mixer is an attribute processing machine. It ingests attributes coming from the proxy and responds by
+triggering calls into a suite of adapters. These adapters in turn communicate with infrastructure
+backends that offer a variety of capabilities. The operator that configures Istio controls what Mixer
```

Comment:
variety of capabilities, such as logging, monitoring, quotas, ACL checking, and more.